### PR TITLE
feat: athena configuration support for Catalog

### LINKED ISSUE #2140: Athena Catalog Configuration
**Is your feature request related to a problem? Please describe.**
Athena configuration does not allow using a custom Catalog (only allows the default aws data catalog)

**Describe the solution you'd like**
We want to be able to specify our own custom catalog.

**Describe alternatives you've considered**
Dump the tables in the default catalog

**Additional context**
Add any other context, documentation links, or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by a7i**
## What does this PR change?
* Provide the ability to set custom Catalog name for Athena

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* Existing users and configuration will not be impacted since it's an optional filed
* New configurations that require a custom catalog can specify this new optional field

## Does this PR address any GitHub or Zendesk issues?
* Closes https://github.com/opencost/opencost/issues/2140

## How was this PR tested?
* Unit Test
* Our dev env of opencost installation running this fork

## Does this PR require changes to documentation?
* Yes but unfortunately Opencost doesn't document any of the Athena configuration ([ref](https://github.com/opencost/opencost-website/blob/main/docs/configuration/AWS.md)) This is only documented for kubecost docs.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

**Comment by mattray** [comment]
Thanks for the PR! I'll raise it with Kubecost Engineering because it looks like it's not an OpenCost issue directly.

**Comment by a7i** [comment]
> Thanks for the PR! I'll raise it with Kubecost Engineering because it looks like it's not an OpenCost issue directly.

@mattray I'm using opencost so it is an OpenCost issue directly in my case.

**Comment by mattray** [comment]
@a7i what are you doing with Athena?

**Comment by a7i** [comment]
> @a7i what are you doing with Athena?

We use it for querying Reserved Instances (RI) and CSP pricing of nodes. By default, OpenCost assumes that all schemas/tables are within the default `AwsDataCatalog` catalog but that is not the case for us. I have seen mentions of this in a kubecost Issue as well (kubecost/cost-analyzer-helm-chart#2290), and hence why I initially linked that.

But this would be a core feature support in OpenCost.

**Comment by lmello** [comment]
Expanding a little bit on @a7i message, to explain our use case: 

Our cur Athena database is in another account and we have a catalog (using Athena and glue) shared with the account where opencost is running, this allow us to do cross account queries on the cur database.

This fix is required so we can specify the catalog name instead of using the AWS default one which does not have cur . 

We keep the default behavior in case the catalog is not specified.

**Review by Sean-Holcomb — APPROVED** [review]
This looks good, thank you for your contribution here

File: pkg/cloud/aws/athenaconfiguration.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ type AthenaConfiguration struct {
 	Bucket     string     `json:"bucket"`
 	Region     string     `json:"region"`
 	Database   string     `json:"database"`
+	Catalog    string     `json:"catalog""`
```
Comment (thomasvn) [inline pkg/cloud/aws/athenaconfiguration.go:15]: Looks like there's an extra `"` here for `"catalog""`. Does this affect the serialization/de-serialization of the JSON?

Comment (thomasvn) [reply pkg/cloud/aws/athenaconfiguration.go:15]: Nvm, this is fixed in #2170 !

**Pull request merged**